### PR TITLE
Part of #17670 Replace Typography with Text component for files: 'review-recovery-phrase' & 'metametrics'

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation
In this PR I have worked on the issue #17670 Replace Typography with Text component and issue #18714 Replace deprecated design system typography consts with enums for the files:

1.  ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js
2.  ui/pages/onboarding-flow/metametrics/metametrics.js

### Before

Screenshot (59)

**COULD NOT GET THE OTHER ONE IN THE UI**

### After

Screenshot (88)

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
Code looks good. Could you please add screenshots to the PR description

**Comment by georgewrmarshall** [comment]
Looks like you might need to update snapshots as well

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 1]
```diff
@@ -77,25 +78,24 @@ export default function OnboardingMetametrics() {
...
+        fontWeight={FontWeight.Bold}
       >
         {t('onboardingMetametricsTitle')}
-      </Typography>
-      <Typography
+      </Text>
+      <Text
         className="onboarding-metametrics__desc"
-        align={TEXT_ALIGN.CENTER}
+        marginTop={2}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:91]: Just curious about where is this style coming from? Previously we did not have margin-top appended

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 2]
```diff
@@ -122,14 +122,15 @@ export default function OnboardingMetametrics() {
               color={IconColor.errorDefault}
             />
             {t('onboardingMetametricsNeverCollect', [
-              <Typography
-                variant={TypographyVariant.span}
+              <Text
+                display={DISPLAY.INLINE_BLOCK}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:126]:
```suggestion
                as="span"
```

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 3]
```diff
@@ -122,14 +122,15 @@ export default function OnboardingMetametrics() {
...
+                variant={TextVariant.bodyMd}
                 key="never"
-                fontWeight={FONT_WEIGHT.BOLD}
+                fontWeight={FontWeight.Bold}
                 marginTop={0}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:130]: You can remove this one now :)

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 4]
```diff
@@ -142,13 +143,14 @@ export default function OnboardingMetametrics() {
               color={IconColor.errorDefault}
             />
             {t('onboardingMetametricsNeverCollectIP', [
-              <Typography
-                variant={TypographyVariant.span}
+              <Text
+                display={DISPLAY.INLINE_BLOCK}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:147]:
```suggestion
                       as="span"
```

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 5]
```diff
@@ -161,29 +163,32 @@ export default function OnboardingMetametrics() {
               color={IconColor.errorDefault}
             />
             {t('onboardingMetametricsNeverSellData', [
-              <Typography
-                variant={TypographyVariant.span}
+              <Text
+                display={DISPLAY.INLINE_BLOCK}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:167]:
```suggestion
         as="span"
```

**Review by georgewrmarshall — COMMENTED** [review]
Hey @PrgrmrHarshShukla, just need to update snapshots and we can get this merged. Thanks!

File: ui/pages/onboarding-flow/metametrics/__snapshots__/metametrics.test.js.snap [hunk 6]
```diff
@@ -7,17 +7,20 @@ exports[`Onboarding Metametrics Component should match snapshot 1`] = `
     data-testid="onboarding-metametrics"
   >
     <h2
-      class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography typography--h2 typography--weight-bold typography--style-normal typography--align-center typography--color-text-default"
+      align="center"
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/metametrics/__snapshots__/metametrics.test.js.snap:10]: Seems like a prop has been incorrectly used and is being passed down to the html element

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 7]
```diff
@@ -77,25 +77,20 @@ export default function OnboardingMetametrics() {
...
       data-testid="onboarding-metametrics"
     >
-      <Typography
-        variant={TypographyVariant.H2}
-        align={TEXT_ALIGN.CENTER}
-        fontWeight={FONT_WEIGHT.BOLD}
+      <Text
+        variant={TextVariant.headingLg}
+        as="h2"
+        align={TextAlign.Center}
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:83]: I believe you want `textAlign` here this is what is being passed down to the html element. docs: https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-text--text-align-story#text-align
```suggestion
        textAlign={TextAlign.Center}
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @PrgrmrHarshShukla, I think there is an incorrect prop being used. `align` should be `textAlign` could you please update

**Review by georgewrmarshall — COMMENTED** [review]
You still have some instances of `align={` in the `Text` components. Please update

**Comment by georgewrmarshall** [comment]
Hey @PrgrmrHarshShukla, just wanted to check in. It looks like this PR is almost there. Did you still want to work on it? If so, that's great! Let us know if you need any assistance or if there are any blockers. If you're unable to continue working on it or haven't had a chance to address it, no worries! I can take it from here and carry it forward. Cheers!

**Comment by georgewrmarshall** [comment]
Hey @PrgrmrHarshShukla, thanks for all your effort with this PR I am going to close it as it has become stale. Please feel free to open another PR in future. Thanks for your contributions thus far!

**Pull request closed without merging**